packages/tenant-process/src/services/tenantService.ts

```diff
@@ -1933,21 +1976,30 @@ function assignDeclaredAttribute(
       type: tenantAttributeType.DECLARED,
       assignmentTimestamp: new Date(),
       revocationTimestamp: undefined,
+      delegationId,
     },
   ];
 }
 
 function reassignDeclaredAttribute(
   attributes: TenantAttribute[],
-  attributeId: AttributeId
+  attributeId: AttributeId,
+  delegationId: DelegationId | undefined
 ): TenantAttribute[] {
   return attributes.map((attr) =>
     attr.id === attributeId
-      ? {
-          ...attr,
-          assignmentTimestamp: new Date(),
-          revocationTimestamp: undefined,
-        }
+      ? attr.type === tenantAttributeType.DECLARED
+        ? {
+            ...attr,
+            assignmentTimestamp: new Date(),
+            revocationTimestamp: undefined,
+            delegationId,
+          }
+        : {
+            ...attr,
+            assignmentTimestamp: new Date(),
+            revocationTimestamp: undefined,
+          }
```

Comment:
This function should be used only for declared attributes so probably this check is not needed, it should always be of type Declared - you could try to refine the type or do some assertion to make sure it compiles